mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1621,28 +1680,35 @@ async fn electrum_request_to(
     client: ElectrumClient,
     request: JsonRpcRequestEnum,
     to_addr: String,
-) -> Result<(JsonRpcRemoteAddr, JsonRpcResponseEnum), String> {
+) -> Result<(JsonRpcRemoteAddr, JsonRpcResponseEnum), JsonRpcError> {
     let (tx, responses) = {
         let connections = client.connections.lock().await;
-        let connection = connections
-            .iter()
-            .find(|c| c.addr == to_addr)
-            .ok_or(ERRL!("Unknown destination address {}", to_addr))?;
+        let connection = connections.iter().find(|c| c.addr == to_addr).ok_or_else(|| {
+            JsonRpcError::new(
+                client.coin_name(),
+                &request,
+                JsonRpcErrorType::Transport(format!("Unknown destination address {}", to_addr)),
```

Comment:
Do you think `JsonRpcErrorType::Internal` is better here? it means that `electrum_request_to` was provided a wrong address which is something internal to the code.

Reply:
`select_ok_sequential` returns server error only if all request fails...so I think `Transport` is better to use than `Internal` here

Reply:
actually it could be internal too.

So which do you think I should I use @shamardy

Reply:
`JsonRpcErrorType::Internal` is the right error here, this error happens if a wrong `to_addr` is provided to the function.